### PR TITLE
#175636846 logout feature

### PR DESCRIPTION
**Pull request by roclena**
**What does this PR do?**
- User logout to end the session
 **Description of Task to be completed?**
- check if the user login in by providing a token in header
- put the token in the blacklist
- check if the token that requested resource is in the blacklist and block it
**How should this be manually tested?**
- `run git clone https://github.com/atlp-rwanda/tech-strikers-bn-backend.git`
- ` run cd tech-strikers-bn-backend`
- `run git checkout ft-logout-#1756368`
- `run npm install`
 - create your .env file from .env.example file and add variable ACCESS_TOKEN_SECRET in that file with your random value
 - add database username, password and database name in your .env file
-` run sequelize db:migrate`
- `run sequelize db:seed:all` in order to have the seeds and apply login on postman
-` run npm run dev`
- use postiman with `localhost:3000/api/V1/user/logout` to logout or end the session
- `run npm run test` to test the login API. It should pass when 
**Any background context you want to provide?**
no
 **What are the relevant pivotal tracker stories?**
- [#175636846]("https://www.pivotaltracker.com/n/projects/2475704")
**Screenshots (if appropriate)**
image
image

**Questions:**

### CODE REVIEW

File: src/controllers/list.controllers.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+        }else{
+            return res.status(401).json("you Loged out?")
+        }
+    }
+    
+
+
+
+}
```
Comment (Duncanwar) [inline src/controllers/list.controllers.js]: add new line

File: src/database/config/tokenlist.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+const whitelist=[];
+export default whitelist;
```
Comment (Duncanwar) [inline src/database/config/tokenlist.js]: add new line

File: src/middlewares/blacklist.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+           }
+
+        }else{
+            return res.status(401).json("invalid session")
+        }
+    }
+
+
+
+}
```
Comment (Duncanwar) [inline src/middlewares/blacklist.js]: add new line

File: src/services/index.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+    static AddToList(data) {
+        list.push(data);
+        return list;
+    }
+    static checklist(data) {
+        return list.includes(data);
+    }
+}
+export default listservices;
```
Comment (Duncanwar) [inline src/services/index.js:13]: export default class listservices and remove that one on line 13.
Remember to add new line

File: src/utils/statusCode.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export default {
+    ok: 200,
+    created: 201,
+    badRequest: 400,
+    unAuthorized: 401,
+    forbidden: 403,
+    notFound: 404,
+    conflict: 409,
+    serverError: 500,
+  };
```
Comment (Duncanwar) [inline src/utils/statusCode.js]: add new line

File: test/logout/controller/user.logout.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      .send(token)
+      .end((req,res) => {
+        const data  = token;
+        expect(res.status).to.equal(200);
+        expect(message1)
+        expect(message1).to.be.equal("Logout success fully");       
+        done();
+      });
+    });
+});
```
Comment (Duncanwar) [inline test/logout/controller/user.logout.js]: add new line

File: test/logout/data/message.js [hunk 7]
```diff
@@ -0,0 +1,4 @@
+export default{
+    message1:"Logout success fully",
+    message2:"Well Connected api"
+}
```
Comment (Duncanwar) [inline test/logout/data/message.js]: add new line

File: test/logout/welcome/initialtest.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+      .end((err, res) => {
+        if (err) done(err);
+        const { message } = res.body;
+        expect(res.status).to.equal(200);
+        expect(message2);
+        expect(message2).to.equal("Well Connected api");
+        done();
+      });
+  });
+});
```
Comment (Duncanwar) [inline test/logout/welcome/initialtest.js]: add new line

**Comment by bigelio** [comment]
You have to do rebase and i will review this pr once you are done.

**Review by Duncanwar — APPROVED** [review]

**Comment by Duncanwar** [comment]
Solve this conflict

**Comment by bigelio** [comment]
Do rebase and we seen if it still working well.

**Review by hervebutera — APPROVED** [review]

**Pull request merged**